packages/knip/src/reporters/gitlab.ts

```diff
@@ -0,0 +1,128 @@
+import { createHash } from 'node:crypto';
+import type { Entries } from 'type-fest';
+import type { Issue, IssueRecords, IssueSeverity, IssueSymbol, Report, ReporterOptions } from '../types/issues.js';
+import { getTitle } from './util.js';
+
+/**
+ * Entry for GitLab Code Quality report. See: https://docs.gitlab.com/ci/testing/code_quality/#code-quality-report-format
+ */
+interface CodeQualityEntry {
+  check_name: string;
+  description: string;
+  severity: 'info' | 'minor' | 'major' | 'critical' | 'blocker';
+  fingerprint: string;
+  location: {
+    path: string;
+    lines: {
+      begin: number;
+    };
+  };
+}
+
+export default async ({ report, issues }: ReporterOptions) => {
+  const entries: CodeQualityEntry[] = [];
+  const hashes = new Set<string>();
+
+  for (const [type, isReportType] of Object.entries(report) as Entries<Report>) {
+    if (!isReportType) {
+      continue;
+    }
+
+    if (type === 'files' || type === '_files') {
+      continue;
+    }
+
+    for (const issue of flatten(issues[type] as IssueRecords)) {
+      const { filePath } = issue;
+
+      if (type === 'duplicates' && issue.symbols) {
+        entries.push(
+          ...issue.symbols.map(symbol => ({
+            check_name: getTitle(type),
+            description: getSymbolDescription({ symbol, parentSymbol: issue.parentSymbol }),
+            severity: convertSeverity(issue.severity),
+            fingerprint: createFingerprint(filePath, symbol.symbol, hashes),
+            location: {
+              path: filePath,
+              lines: {
+                begin: symbol.line ?? 0,
+              },
+            },
+          }))
+        );
+      } else {
+        entries.push({
+          check_name: getTitle(type),
+          description: getIssueDescription(issue),
+          severity: convertSeverity(issue.severity),
+          fingerprint: createFingerprint(filePath, issue.symbol, hashes),
+          location: {
+            path: filePath,
+            lines: {
+              begin: issue.line ?? 0,
+            },
+          },
+        });
+      }
+    }
+  }
+
+  const output = JSON.stringify(entries);
+
+  // See: https://github.com/nodejs/node/issues/6379
+  // @ts-expect-error _handle is private
+  process.stdout._handle?.setBlocking?.(true);
+  process.stdout.write(`${output}\n`);
+};
+
+function flatten(issues: IssueRecords): Issue[] {
+  return Object.values(issues).flatMap(Object.values);
+}
+
+function convertSeverity(severity?: IssueSeverity): CodeQualityEntry['severity'] {
+  switch (severity) {
+    case undefined:
+      return 'info';
```

Comment:
Maybe instead of this initial `undefined` case we could move it to the end as the default case?